resources/assets/js/annotations/ol/snapInteraction.js

```diff
@@ -32,17 +32,13 @@ export default {
             // Return always true to enable snapping when first point is set
             return true;
         },
-        updateSnapCoordsWithBoundaryRestriction(mapBrowserEvent) {
-            // Coordinates need to be set here, because only this event contains the last set coordinate
-            this.snappingCoords = mapBrowserEvent.coordinate;
-            this.shouldSnap = mapBrowserEvent.originalEvent.ctrlKey;
-            
+        maybeRestrictToBounds(mapBrowserEvent) {
             // If boundary restriction is enabled, check if coordinates are within bounds
             if (this.restrictToBounds) {
                 return this.drawsOnImage;
```

Comment:
This should not be a computed property as it is recomputed each time the event is fired anyway. Also it still relies on `updateSnapCoords` which should be decoupled from this logic.